lib/type_meme.dart

```diff
@@ -0,0 +1,62 @@
+import 'package:flutter/material.dart';
+
+class TypeMeme extends StatefulWidget {
+  @override
+  State<StatefulWidget> createState() => new TypeMemeState();
+}
+
+// Represents the states of typing text onto an image to make a meme.
+class TypeMemeState extends State<TypeMeme> {
+  InputValue _currentMessage = InputValue.empty;
+
+  @override
+  Widget build(BuildContext context) {
+    return new Scaffold(
+        appBar: new AppBar(title: new Text('Make yo meme')),
+        body: new Column(children: <Widget>[
+          new Stack(children: [
+            new Image.asset('assets/test_image.jpg'),
+            new Text(_currentMessage.text,
+                textAlign: TextAlign.center,
```

Comment:
When I run this, it looks like this TextAlign is making it so that when the text extends to multiple lines it is centered, but a single line of text is still left-aligned. If you want it to consistently be centered, try adding the following parameter to the Stack: `alignment: FractionalOffset.topCenter`

I believe what's happening here is that the textAlign parameter is aligning the text within the space allotted for it, whereas the stack alignment parameter is aligning all stack children within the space allotted for the full stack. Here, when the text is shorter than a line, the width allotted for it is basically just the width of the text, so changing the textAlign doesn't do anything. When the text is longer than a line, the width is the full width of the stack, and textAlign makes a visible difference on any line that isn't full-length. By the same token, when the text width is the full width of the stack, the stack alignment parameter won't do anything, but when it's shorter it has room to make a difference.